pmlc/dialect/xsmm/ir/ops.cc

```diff
@@ -274,9 +274,38 @@ static std::string stringfy(BinaryKind kind) {
 //
 // --- BRGemmInvokeF32Op ---
 //
+BRGemmInvokeF32Op::operand_range BRGemmInvokeF32Op::getOperandsForA() {
+  auto aType = a().getType().cast<MemRefType>();
+  auto cType = c().getType().cast<MemRefType>();
+  return getOperands().slice(4 + cType.getRank(), aType.getRank());
+}
+
+BRGemmInvokeF32Op::operand_range BRGemmInvokeF32Op::getOperandsForB() {
+  auto aType = a().getType().cast<MemRefType>();
+  auto bType = b().getType().cast<MemRefType>();
+  auto cType = c().getType().cast<MemRefType>();
+  return getOperands().slice(4 + cType.getRank() + aType.getRank(),
+                             bType.getRank());
+}
+
+BRGemmInvokeF32Op::operand_range BRGemmInvokeF32Op::getOperandsForC() {
+  auto cType = c().getType().cast<MemRefType>();
+  return getOperands().slice(4, cType.getRank());
+}
 
 void BRGemmInvokeF32Op::print(OpAsmPrinter &p) {
-  // assert(0 && "implement me");
+  auto funcType = FunctionType::get(
```

Comment:
why not using the assembly form here? https://mlir.llvm.org/docs/OpDefinitions/#declarative-assembly-format

Reply:
I guess this fine for now as all the other prints are done in the same way... e.g. when you go to line 157 in the same finel.

Reply:
Fair enough